packages/x-date-pickers/src/StaticDatePicker/StaticDatePicker.types.ts

```diff
@@ -20,7 +20,7 @@ export interface StaticDatePickerSlotsComponentsProps<TDate>
 
 export interface StaticDatePickerProps<TDate>
   extends BaseDatePickerProps<TDate>,
-    MakeOptional<StaticOnlyPickerProps, 'displayStaticWrapperAs'> {
+    StaticOnlyPickerProps {
```

Comment:
@flaviendelangle The usage of `MakeOptional` here was causing the `onClose` to lose the `deprecated` styling in VsCode IDE.
What's the case with WebStorm? 🤔 

**Before:**
<img width="544" alt="Screenshot 2023-02-23 at 14 57 29" src="https://user-images.githubusercontent.com/4941090/220912943-8a2ec9bf-3778-43e1-8219-c3867da4a6f3.png">

**After:**
<img width="626" alt="Screenshot 2023-02-23 at 14 57 49" src="https://user-images.githubusercontent.com/4941090/220913019-550a7545-ce63-4a73-b18e-c54da846ac07.png">